pages/agent/v3/hooks.md

```diff
@@ -147,6 +147,11 @@ they are run as part of each job:
 | `pre-exit`      | <span class="add-icon-agent">Agent</span><br /><span class="add-icon-repository">Repository</span><br /><span class="add-icon-plugin">Plugin (non-vendored)</span><br /><span class="add-icon-plugin">Plugin (vendored)</span> | Runs before the job finishes. Useful for performing cleanup tasks. |
 {: class="table table--no-wrap"}
 
+> 📘 Hook failure behavior
+> If any hooks fail with a non-zero exit code, the job will be aborted at that point.
+> The behavior for most `post-*` hooks above, however, is a little different. If a `command` is the core of a job, the `command` job would have already run and completed before the `post-*` hook is run. Therefore, a `post-*` hook's success or failure has no impact on stopping the core `command` job from running.
+> The exception to this is the `post-checkout` hook, which runs after checkout but before the core `command` job runs. In such circumstances, if the `post-checkout` hook fails, then the `command` job will not run.
```

Comment:
The wording here is weird, since you say "if any hooks fail ..." which sounds quite global. Then next sentence contradicts it.

This also gives a bit too much context in my opinion. Most people care about the what/how and not the _why_.

So maybe something like:
```suggestion
> For most hooks, if they fail with a non-zero exit code, the job will be aborted at that point.
> For `post-*` hooks (except for `post-checkout`), if they fail with a non-zero exit code, the job has no impact.
```

Reply:
Thanks for your feedback @thejcannon .

I agree with you in re-writing this note/callout to front-load the _what/how_ explanation, since in this context, it's the most important explanation about 'Hook failure behavior'.
However, I would also argue that most people who visit documentation _are_ interested in the _why_ too, and that people consult documentation to look out for these explanations, so that they don't have to invest extra time in experimenting with a product to work out the 'why' through trial and error.

I will re-write this note/callout to front-load your suggestion, but I will retain the explanation for the benefit of other readers who'd like a more detailed explanation on why.

Reply:
Here's a preview of the final result:
https://2883--bk-docs-preview.netlify.app/docs/agent/v3/hooks#hook-failure-behavior